projects/packages/waf/src/class-waf-request.php

```diff
@@ -334,17 +334,21 @@ public function get_get_vars() {
 	 * @return array{string, scalar}[]
 	 */
 	public function get_post_vars() {
+		$content_type = $this->get_header( 'content-type' );
 		if ( ! empty( $_POST ) ) {
+			// If $_POST is populated, use it.
 			return flatten_array( $_POST );
-		} elseif ( strpos( $this->get_header( 'content-type' ), 'application/json' ) !== false ) {
+		} elseif ( strpos( $content_type, 'application/json' ) !== false ) {
 			// Attempt to decode JSON requests.
 			$decoded_json = json_decode( $this->get_body(), true ) ?? array();
 			return flatten_array( $decoded_json, 'json', true );
-		} else {
-			// Attempt to retrieve all parameters when method used isn't POST
-			$body = $this->get_body();
-			parse_str( $body, $params );
+		} elseif ( 'application/x-www-form-urlencoded' === $content_type ) {
```

Comment:
```suggestion
		} elseif ( strpos( $content_type, 'application/x-www-form-urlencoded' ) !== false ) {
```
I know there are cases where the `content-type` may be set to a value such as:
```
Content-Type: application/x-www-form-urlencoded; charset=utf-8
```
This is why we used `strpos` in the JSON check, could we use the same thing here?